### PR TITLE
[MAINT] Removed reference fields from CPacket pinned to the SRT header

### PR DESCRIPTION
**Pull request by ethouris**
Reference fields like `m_iSeqNo` that were set to the value in the objects self parts of the array keeping the header have been turned into methods.

Only `m_pcData` is left unchanged because this is much more complicated.

### CODE REVIEW

File: test/test_buffer_rcv.cpp [hunk 1]
```diff
@@ -54,22 +54,27 @@ class CRcvBufferReadMsg
...
-            packet.m_iMsgNo |= PacketBoundaryBits(PB_FIRST);
+            pktMsgFlags |= PacketBoundaryBits(PB_FIRST);
         if (pb_last)
-            packet.m_iMsgNo |= PacketBoundaryBits(PB_LAST);
+            pktMsgFlags |= PacketBoundaryBits(PB_LAST);
+
+        if (!out_of_order)
+        {
+            pktMsgFlags |= MSGNO_PACKET_INORDER::wrap(1);
+        }
```
Comment (maxsharabayko) [inline test/test_buffer_rcv.cpp:73]: Those braces were not used in the above if-statements.
```suggestion
            pktMsgFlags |= MSGNO_PACKET_INORDER::wrap(1);
```

File: test/test_buffer_rcv.cpp [hunk 2]
```diff
@@ -54,22 +54,27 @@ class CRcvBufferReadMsg
...
+        packet.set_msgflags(pktMsgFlags);
```
Comment (maxsharabayko) [inline test/test_buffer_rcv.cpp:74]: Repeating `if (!out_of_order)` below. Maybe a merge issue. Can be combined, I belive.

**Review by maxsharabayko — COMMENTED** [review]
Consider minor comments.
I am a bit concerned the naming convention is not followed, i.e. `set_timestamp` instead of `setTimestamp` function names are used. I guess that might be tricky to do `set` with upper case, but `get` without the `get` itself, only using the work starting with lowercase.

File: test/test_buffer_rcv.cpp (hunk 2 above)
Comment (ethouris) [reply test/test_buffer_rcv.cpp:74]: No. This is intended and the order matters. This call to set_msgflags influences the value being checked in the next expector.

File: test/test_buffer_rcv.cpp (hunk 1 above)
Comment (ethouris) [reply test/test_buffer_rcv.cpp:73]: Could be. Actually this is a test, it doesn't exactly matter.

**Review by maxsharabayko — APPROVED** [review]

**Pull request merged**